src/client/components/UserInput.tsx

```diff
@@ -134,8 +155,14 @@ class UserInput extends React.Component<Props, State> {
 		}
 	}
 
-	public textInputChange = (event: any) => {
-		this.props.onMessageDraftChange('TEXT', event);
+	public textInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+		if (!this.props.disableBackspace || event.target.value.length > this.props.value.textContent.length) {
```

Comment:
Also check that `event.target.value` starts with `this.props.value.textContent`. Otherwise some developer console fiddling would still allow me to change, for example, the message draft "aijai" to "oijoijoi".